dsm/resource_app.go

```diff
@@ -189,9 +216,82 @@ func resourceUpdateApp(ctx context.Context, d *schema.ResourceData, m interface{
 			})
 			return diags
 		}
-		return resourceReadApp(ctx, d, m)
 	}
-	return nil
+    //Modified by Ravi Gopal
+	app_object := make(map[string]interface{})
+
+	if d.HasChange("default_group") {
+		if default_group := d.Get("default_group").(string); len(default_group) > 0 {
+			app_object["default_group"] = d.Get("default_group")
+		}
+	}
+	if d.HasChange("other_group") {
+		old_group, new_group := d.GetChange("other_group")
+		/*compares the old and new state
+		 * and seggregtes the new groups and groups that are to be deleted.
+		*/
+		add_group_ids, del_group_ids := form_add_and_del_groups(old_group, new_group)
+		//Add the groups to be deleted
+		if len(del_group_ids) > 0 {
+			app_object["del_groups"] = del_group_ids
+		}
+		//Add the new groups
+		if len(add_group_ids) > 0 {
+			add_group_perms := form_group_permissions(d.Get("other_group_permissions"))
+			app_add_group := make(map[string]interface{})
+			for i := 0; i < len(add_group_ids); i++ {
+				if perms, ok := add_group_perms[add_group_ids[i]]; ok {
+					app_add_group[add_group_ids[i]] = perms
+				}else{
+					app_add_group[add_group_ids[i]] = []string{"SIGN", "VERIFY", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "DERIVEKEY", "MACGENERATE", "MACVERIFY", "EXPORT", "MANAGE", "AGREEKEY", "AUDIT"}
+				}
+			}
+			app_object["add_groups"] = app_add_group
+		}
+	}
+	if d.HasChange("description") {
+		app_object["description"] = d.Get("description")
+	}
+	//Modifies the existing groups
+	if d.HasChange("mod_group_permissions"){
+		if mod_group := d.Get("mod_group_permissions").(map[string]interface{}); len(mod_group) > 0 {
+			mod_group := d.Get("mod_group_permissions").(map[string]interface{})
+			app_mod_group := make(map[string]interface{})
+			//if default_group has changes in permissions
+			default_group := d.Get("default_group").(string)
+			if perms, ok := mod_group[default_group]; ok {
+				app_mod_group[default_group] = strings.Split(perms.(string), ",")
+			}
+			//checking whether all the group_ids from mod_group_permissions exists in other groups or not
+			//if not it will ignore the mod_group_permissions of the unavailable group_id
```

Comment:
I don't like silently ignoring logic errors

Reply:
Now it will not ignore the errors.
It will return an error.

Ex:

│ Following group_ids are not available in other_group:
│  [0fa3e2ff-6798-4720-bb60-c86fd4dd4570 daeb1fe9-f908-4583-9d59-456ad182b4bc]